src/internal/obj/amazon_client.go

```diff
@@ -272,23 +274,24 @@ func newAmazonClient(region, bucket string, creds *AmazonCreds, cloudfrontDistri
 	return awsClient, nil
 }
 
-func (c *amazonClient) Writer(ctx context.Context, name string) (io.WriteCloser, error) {
-	if c.advancedConfig.Reverse {
-		name = reverse(name)
-	}
-	return newBackoffWriteCloser(ctx, c, newWriter(ctx, c, name)), nil
+func (c *amazonClient) Put(ctx context.Context, name string, r io.Reader) (retErr error) {
+	defer func() { retErr = c.transformError(retErr, name) }()
+	ctx, cf := context.WithCancel(ctx)
+	defer cf()
+	_, err := c.uploader.UploadWithContext(ctx, &s3manager.UploadInput{
+		ACL:             aws.String(c.advancedConfig.UploadACL),
+		Body:            r,
+		Bucket:          aws.String(c.bucket),
+		Key:             aws.String(name),
+		ContentEncoding: aws.String("application/octet-stream"),
+	})
+	return err
 }
 
-func (c *amazonClient) Walk(_ context.Context, name string, fn func(name string) error) error {
+func (c *amazonClient) Walk(_ context.Context, name string, fn func(name string) error) (retErr error) {
```

Comment:
We probably should add TODOs for the implementations that do not respect the context.